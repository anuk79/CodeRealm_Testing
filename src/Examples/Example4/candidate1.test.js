
// vendors
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// components
import Candidate from './candidate';

// mocking react redux library
jest.mock('react-redux', () => ({
    connect: () => {
        return (component) => {
            return component
        };
    }
}));

describe('Candidate component', () => {
    const mockFetchCandidateDetails = jest.fn();
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Candidate isDataFetching={true} fetchCandidateDetails={mockFetchCandidateDetails}
        />);
    });
    it('should call fetchCandidateDetails', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(mockFetchCandidateDetails).toBeCalled();
    });
    
    it('should render correctly when data is isDataFetching', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('LoadingSpinner').length).toBe(1);
    });

    it('should render correctly when some error', () => {
        wrapper.setProps({
            isDataFetching: false,
            hasError: true
        });
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('.candidate-error').length).toBe(1);
    });

    it('should render correctly when no error and data isDataFetching is done', () => {
        wrapper.setProps({
            isDataFetching: false,
            hasError: false,
            userDetails: { name: 'test name' }
        })
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('CandidateView').length).toBe(1);
        expect(wrapper.find('CandidateView').prop('userDetails')).toEqual({ name: 'test name' });
    });
});