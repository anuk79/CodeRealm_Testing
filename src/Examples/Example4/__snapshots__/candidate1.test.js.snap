// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candidate component should call fetchCandidateDetails 1`] = `
<LoadingSpinner
  customClassName="candidate-loading-spinner"
/>
`;

exports[`Candidate component should render correctly when data is isDataFetching 1`] = `
<LoadingSpinner
  customClassName="candidate-loading-spinner"
/>
`;

exports[`Candidate component should render correctly when no error and data isDataFetching is done 1`] = `
<CandidateView
  candidateDetails={
    Object {
      "name": "test name",
    }
  }
/>
`;

exports[`Candidate component should render correctly when some error 1`] = `
<div
  className="candidate-error"
>
  Some error occurred. Please try again later.
</div>
`;
